extensions/ql-vscode/src/databases/local-databases-ui.ts

```diff
@@ -962,27 +978,67 @@ export class DatabaseUI extends DisposableObject {
    */
   private async chooseAndSetDatabase(
     byFolder: boolean,
+    subFolder: boolean,
     progress: ProgressCallback,
-  ): Promise<DatabaseItem | undefined> {
+  ): Promise<DatabaseItem[] | DatabaseItem | undefined> {
     const uri = await chooseDatabaseDir(byFolder);
     if (!uri) {
       return undefined;
     }
 
-    if (byFolder && !uri.fsPath.endsWith("testproj")) {
-      const fixedUri = await this.fixDbUri(uri);
-      // we are selecting a database folder
-      return await this.databaseManager.openDatabase(fixedUri, {
-        type: "folder",
-      });
+    if (subFolder) {
+      if (!byFolder) {
+        return undefined;
+      }
+
+      const databases: DatabaseItem[] = [];
+      const failures: string[] = [];
+      const entries = await workspace.fs.readDirectory(uri);
+      for (const entry of entries) {
+        if (entry[1] === FileType.Directory) {
+          try {
+            const fixedUri = await this.fixDbUri(Uri.joinPath(uri, entry[0]));
+            const database = await this.databaseManager.openDatabase(fixedUri, {
+              type: "folder",
+            });
+            databases.push(database);
+          } catch (e) {
+            failures.push(entry[0]);
+          }
+        }
+      }
+
+      if (failures.length) {
+        void showAndLogErrorMessage(
+          this.app.logger,
+          `Failed to import ${failures.length} database(s) (${failures.join(
```

Comment:
What gets printed here? Is this the full path to the database? If so, this would be too long to show in the popup.

Can you use the `fullMessage` option to contain all of the database path names, and just the number of failures in the `message` argument?

Reply:
Sounds good to me! Didn't know there's that argument.